parser-typechecker/src/Unison/Codebase/Branch.hs

```diff
@@ -211,80 +220,163 @@ branch0 terms types children edits =
 -- | Derive the 'deepTerms' field of a branch.
 deriveDeepTerms :: Branch0 m -> Branch0 m
 deriveDeepTerms branch =
-  branch {deepTerms = makeDeepTerms (_terms branch) (nonEmptyChildren branch)}
+  branch {deepTerms = R.fromList (makeDeepTerms branch)}
```

Comment:
Just curious: was memory usage (or something) worse with a `Relation Referent Name` accumulator, rather than this `[(Referent, Name)]` accumulator that's `R.fromList`'d at the end?

Reply:
We suspected that at least for the `Relation4` in `deep*Metadata` that it was worse to use the `R4` as an accumulator, but I don't remember if we tested it. I guess that `Relation` wouldn't be worse than `[(,)]`, but maybe it slipped in as symmetry? I figure it also wouldn't be better.
@tstat Do you remember a better answer?